src/main/java/io/github/thebusybiscuit/sensibletoolbox/items/MultiBuilder.java

```diff
@@ -532,21 +532,25 @@ public void run() {
                 b.setType(rec.target, true);
 
                 // queue up the next set of blocks
-                if (rec.layersLeft > 0) {
-                    for (int x = -1; x <= 1; x++) {
-                        for (int y = -1; y <= 1; y++) {
-                            for (int z = -1; z <= 1; z++) {
-                                Block b1 = b.getRelative(x, y, z);
-
-                                if ((x != 0 || y != 0 || z != 0) && b1.getType() == rec.source && STBUtil.isExposed(b1)) {
-                                    queue.offer(new SwapRecord(rec.player, b1, rec.source, rec.target, rec.layersLeft - 1, rec.builder, slot, rec.chargeNeeded));
-                                }
+                queueNextSet(rec, b, slot);
+
+                didWork = true;
+            }
+        }
+
+        private void queueNextSet(SwapRecord rec, Block b, int slot){
+            if (rec.layersLeft > 0) {
+                for (int x = -1; x <= 1; x++) {
+                    for (int y = -1; y <= 1; y++) {
+                        for (int z = -1; z <= 1; z++) {
+                            Block b1 = b.getRelative(x, y, z);
+
+                            if ((x != 0 || y != 0 || z != 0) && b1.getType() == rec.source && STBUtil.isExposed(b1)) {
+                                queue.offer(new SwapRecord(rec.player, b1, rec.source, rec.target, rec.layersLeft - 1, rec.builder, slot, rec.chargeNeeded));
```

Comment:
Queue#offer() returns a boolean that determines whether the operation was successful or not.
This shouldn't be ignored, so I'd check that and return if no more items were able to be added.

Reply:
What do you mean "return if no more items were able to be added"? I can make the method a bool and return false if Queue#offer() returns false but then I'd have no real use for it in run() just above

Reply:
I meant that you should probably do `return;` to break out of the loop if the queue cannot accept any more items, since we can basically stop at that point.

Reply:
Aaah lol yea that would be wise

Reply:
It makes for a long ass if clause, any idea how to make it prettier?